### PR TITLE
SONARPY-978 Rule S6397: Character classes in regular expressions shou…

### PR DESCRIPTION
**Pull request by marco-bearzi-sonarsource**
…ld not contain only one character

### CODE REVIEW

File: python-checks/src/main/java/org/sonar/python/checks/regex/SingleCharCharacterClassCheck.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+@Rule(key = "S6397")
+public class SingleCharCharacterClassCheck extends AbstractRegexCheck {
+
+  @Override
+  public void checkRegex(RegexParseResult regexParseResult, CallExpression regexFunctionCall) {
+    Optional.ofNullable(regexFunctionCall.calleeSymbol())
+      .flatMap(symbol -> Optional.ofNullable(symbol.fullyQualifiedName()))
+      .filter(fqn -> lookedUpFunctions().containsKey(fqn))
+      .filter(fqn -> !getFlagSet(regexFunctionCall, fqn).contains(Pattern.COMMENTS))
```
Comment (nils-werner-sonarsource) [inline python-checks/src/main/java/org/sonar/python/checks/regex/SingleCharCharacterClassCheck.java:37]:
```suggestion
      .filter(fqn -> !regexParseResult.getResult().activeFlags().contains(Pattern.COMMENTS))
```
Would be an alternative to not expose the getFlagSet function.

File: python-checks/src/test/resources/checks/regex/singleCharCharacterClassCheck.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    changed = re.match(r"a[\]a", input)
+    changed = re.match(r"a a", input)
+
+    changed = re.compile(r'[ \t # comment]', re.X)
+
+    changed = re.compile(r'[ \t  ]', re.X)
+    changed = re.compile(r'[ \t]', re.X)
+    changed = re.match(r'[ \t]', input, re.X)
+
+    # TODO : False Negatives. We deactivated the SingleCharCharacterClassFinder whenever the flag X or VERBOSE is set.
```
Comment (nils-werner-sonarsource) [inline python-checks/src/test/resources/checks/regex/singleCharCharacterClassCheck.py:55]: We could connect this TODO to the corresponding analyzer-commons [ticket](https://github.com/SonarSource/sonar-analyzer-commons/issues/217)

**Review by nils-werner-sonarsource — APPROVED** [review]
LGTM! See comments

File: python-checks/src/main/java/org/sonar/python/checks/regex/SingleCharCharacterClassCheck.java (hunk 1 above)
Comment (andrea-guarino-sonarsource) [reply python-checks/src/main/java/org/sonar/python/checks/regex/SingleCharCharacterClassCheck.java:37]: Indeed, thanks!

**Pull request merged**